test/unit/lbp-wrapper.js

```diff
@@ -100,32 +84,30 @@ describe("Contract: LBPWrapper", async () => {
             SYMBOL,
             tokenAddresses,
             WEIGHTS,
-            WEIGHTS,
             swapsEnabled,
             startTime,
             endTime,
-            END_WEIGHTS,
-            setup.roles.prime.address,
-            initUserData
+            END_WEIGHTS
           )
       ).to.be.revertedWith("LBPWrapper: only owner function");
     });
     it("$ success", async () => {
-      await setup.lbpWrapper
+      const tx = await setup.lbpWrapper
         .connect(setup.roles.root)
         .deployLbpFromFactory(
           NAME,
           SYMBOL,
           tokenAddresses,
           WEIGHTS,
-          WEIGHTS,
           swapsEnabled,
           startTime,
           endTime,
-          END_WEIGHTS,
-          setup.roles.prime.address,
-          initUserData
+          END_WEIGHTS
         );
+      const receipt = await tx.wait();
+      setup.Lbp = setup.Lbp.attach(await setup.lbpWrapper.lbp());
```

Comment:
here, `setup.LBP` is being overwritten. Is this what we wanted?

Reply:
My understanding was that we attach the instance of LBP that has just been successful created to `setup.Lbp`

Reply:
Nope, it returns an instance of connected contract instance.